### ISSUE TITLE
Add ability for aws_glue_crawler to use catalog table targets

### ISSUE THREAD

**Issue by dkuntz2** [terraform-providers/terraform-provider-aws#8631]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

The Glue `create-crawler` api call allows for using `CatalogTargets`, which allow crawling already created glue tables. The current target types terraform supports only allow the reverse, crawling data to find tables.

Unfortunately this means the glue crawler can and will do a number of unexpected things for sometimes unknown reasons, specifically creating a bunch of tables which should just be partitions of existing tables.

Allowing the crawler to be configured to crawl tables instead would prevent this, and only update existing tables.

### New or Affected Resource(s)

- `aws_glue_crawler`

### Potential Terraform Configuration

```hcl
# ideally this would look something like
resource "aws_glue_crawler" "crawler" {
  catalog_target {
    database_name = "database"
    tables = [
      "list",
      "of",
      "tables",
    ]
  }
}
```

### References

* [CreateCrawler api reference](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-crawler-crawling.html#aws-glue-api-crawler-crawling-CreateCrawler)
* [CatalogTarget api reference](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-crawler-crawling.html#aws-glue-api-crawler-crawling-CrawlerTargets)

**Comment by dkuntz2** [comment]
As an additional note, I would be willing to try to add this feature if desired, but won't be able to do any real work on it until next week.

**Comment by bflad** [comment]
Support for this has been merged and will release with version 2.23.0 of the Terraform AWS Provider, in a week. 👍 Thanks to @azclub for the implementation.
